subprojects/platform-jvm/src/main/java/org/gradle/jvm/internal/resolve/SourceSetDependencyResolvingClasspath.java

```diff
@@ -38,12 +39,13 @@ public SourceSetDependencyResolvingClasspath(
         List<ResolutionAwareRepository> remoteRepositories,
         AttributesSchemaInternal attributesSchema,
         ImmutableModuleIdentifierFactory moduleIdentifierFactory,
-        BuildOperationExecutor buildOperationExecutor) {
+        BuildOperationExecutor buildOperationExecutor,
+        ImmutableAttributesFactory attributesFactory) {
         super(binarySpec,
             "source set '" + sourceSet.getDisplayName() + "'",
             dependencyResolver,
             remoteRepositories,
-            new JvmLibraryResolveContext(binarySpec.getId(), binaryVariants, dependencies, UsageKind.API, sourceSet.getDisplayName(), moduleIdentifierFactory), attributesSchema, buildOperationExecutor);
+            new JvmLibraryResolveContext(binarySpec.getId(), binaryVariants, dependencies, UsageKind.API, sourceSet.getDisplayName(), moduleIdentifierFactory), attributesSchema, buildOperationExecutor, attributesFactory);
```

Comment:
We need to get rid of this crap. We're never going to extend the JVM software model, so it should be purged.